### ISSUE TITLE
Game exit on "show desktop" in windowed mode

### ISSUE THREAD

**Issue by Dawn-Xu-helloworld** [Revolutionary-Games/Thrive#3958]
**To Reproduce**
Steps to reproduce the behaviour:
1. Start the game in windowed mode
2. press Win+D or "show desktop" on task bar
4.The game crashed and exit without any errors and error logs

**System info (please complete the following information):**
- OS: [Windows 10 22H2]
- Graphics card[OpenGL ES 3.0 Renderer: AMD Radeon (TM) R9 Fury Series]
- Thrive version [e.g.0.600]

**Comment by 84634E1A607A** [comment]
This is 99.9% an engine bug. I don't really think that we can do anything other than waiting for one day, with an update of Godot, this problem vanishes.

**Comment by hhyyrylainen** [comment]
I agree that this is basically guaranteed to be a Godot engine bug. Have you tried out the Godot Editor or another Godot game to see if the same problem happens? Also does this happen each time on your computer or did you just see this once?

**Comment by Dawn-Xu-helloworld** [comment]
> I agree that this is basically guaranteed to be a Godot engine bug. Have you tried out the Godot Editor or another Godot game to see if the same problem happens? Also does this happen each time on your computer or did you just see this once?

It's seems that that won't happen in other Godot games

**Comment by hhyyrylainen** [comment]
It could perhaps be caused by the fact that Thrive defaults to full screen mode, and windowed mode is applied once settings are loaded (this sadly needs to be done as there doesn't seem to be another way in Godot). Which could be the difference why Thrive is behaving like this. This is probably one of the things that will improve with Godot 4.0 as they've overhauled their window management code so we might be able to do things better or this might just not be a problem in Godot 4.

**Comment by hhyyrylainen** [comment]
I'll assume that the window handling in Godot 4 is sufficiently different that this is either fixed or needs to be reported again anyway.
